### PR TITLE
Updating ghodss/yaml and gopkg.in/yaml.v2 to latest version 2

### LINKED ISSUE #67477: duplicate container volumeMounts not validated
I accidentally included a duplicate `volumeMounts:` line below, and validation did not fail. I expected this to fail:

```
apiVersion: v1
kind: Pod
metadata:
  name: bugtest
spec:
  restartPolicy: Never
  containers:
    - image: debian:stretch
      name: rbd-rw
      command: ["bash", "-c", "cat /proc/mounts | grep /mnt"]
      volumeMounts:
      - name: foo
        mountPath: /mnt/foo
      volumeMounts:
      - name: bar
        mountPath: /mnt/bar
  volumes:
    - name: foo
      emptyDir: {}
    - name: bar
      emptyDir: {}
```

But nope, no error on apply, and only one of the volumes shows up:

```
dev:~ $ kubectl apply -f test.yaml
pod/bugtest created
dev:~ $ kubectl logs bugtest
/dev/mapper/debvg-root /mnt/bar ext4 rw,relatime,errors=remount-ro,data=ordered 0 0
```

Reproduced with client+server Kubernetes 1.11.2, and also with the kubectl client at git tip (cffa2aed0e4f3a3723d1cf977e3386dbeb1aa046):

```
dev:~ $ kubectl version
Client Version: version.Info{Major:"", Minor:"", GitVersion:"v0.0.0-master+$Format:%h$", GitCommit:"$Format:%H$", GitTreeState:"", BuildDate:"1970-01-01T00:00:00Z", GoVersion:"devel +015873c153 Mon Aug 13 04:48:19 2018 +0000", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-07T23:08:19Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

Do we just need to set the YAML package's [Decoder.SetStrict](https://godoc.org/gopkg.in/yaml.v2#Decoder.SetStrict) to true?

I see no references to SetStrict in the git repo.

Will investigate later. Filing before I forget.

/cc @lavalamp

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

The versions of ghodss/yaml and gopkg.in/yaml.v2 we have vendored are very very old.. We need to move up to latest versions. So far the main changes that shows up in test cases seems to be
- Tighten up the serialization of time stamp with quotes
- Better line numbers

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This does not attempt to fix https://github.com/kubernetes/kubernetes/issues/67477 as there are changes in progress in ghodss/yaml. This PR is just to get whatever changes are currently upstream into our repo

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by cblecker** [comment]
I'm good with the dep bump, but I'd like a review from the cli folks on the test data changes.

/cc @kubernetes/sig-cli-pr-reviews

File: pkg/printers/internalversion/printers_test.go
Comment (sttts) [inline pkg/printers/internalversion/printers_test.go]: what changed here in the type?

Comment (dims) [reply pkg/printers/internalversion/printers_test.go]: `func Unmarshal(y []byte, o interface{}, opts ...JSONOpt) error` is the new method signature. please see vendor/github.com/ghodss/yaml/yaml.go

**Comment by smarterclayton** [comment]
This only affects serialization of timestamps, not deserialization?

**Comment by dims** [comment]
@smarterclayton correct!

**Comment by dims** [comment]
@smarterclayton @sttts i tracked it down to this change:
https://github.com/go-yaml/yaml/commit/1f2a25ba9402c70a7806e84531ef763943739072#diff-3df5f85b48572bebfd409c300c38f308R311

**Comment by smarterclayton** [comment]
This looks ok to me.  I can approve it if someone does the review.  I think I agree with it, but because serialization is so important I want sttts or someone else to comment.

**Comment by dims** [comment]
Thanks @smarterclayton i'll work with @sttts tomorrow and get it ready.

**Comment by lavalamp** [comment]
Having looked at the vendor changes, my only concern is what new yaml files are we supporting? Are people likely to start using some new yaml feature that breaks on them if apiserver is rolled back? Seems low risk.

/approve

Seems like you should still get someone from SIG CLI to look at the test data changes.

**Comment by cblecker** [comment]
/hold

Just to make sure that @sttts and @dims are able to sync up before merge.

**Comment by sttts** [comment]
Looks innocent, especially with @dims's additional test of unmarashaling without quotes. From yaml point of view, using quotes is valid. So it lgtm from my side.

Giving @soltysh a chance to take a look.

**Comment by dims** [comment]
@sttts thanks! adding this early in the cycle, so we can catch any regressions

**Review by soltysh — APPROVED** [review]
/lgtm

**Comment by soltysh** [comment]
I guess I'm the last one to say it's ok. So based on the fast that Stefan had a change to look at it and all other are ok. I'm adding my 2 cli cents and
/hold cancel

**Pull request merged**
